folly/io/async/AsyncSSLSocket.cpp

```diff
@@ -1359,6 +1397,63 @@ void AsyncSSLSocket::handleConnect() noexcept {
   AsyncSocket::handleInitialReadWrite();
 }
 
+#if FOLLY_OPENSSL_HAS_TLS13
+int AsyncSSLSocket::handleReadEarlyData() noexcept {
+  VLOG(3) << "AsyncSSLSocket::handleReadEarlyData() this=" << this
+          << ", fd=" << fd_ << ", state=" << int(state_) << ", "
+          << "sslState=" << sslState_ << ", events=" << eventFlags_;
+  size_t bufSize = SSL_CTX_get_max_early_data(ctx_->getSSLCtx());
+  unsigned char buf[bufSize];
```

Comment:
Variable length arrays are discouraged (especially since `SSL_set_max_early_data` blindly accepts a value).

One concrete reason is that AsyncSSLSocket::handleAccept() can be called under a fiber context, where the fiber's stack size can be fairly small. 

Unfortunately, I don't think it's guaranteed that the application's read callback is set at this point (normally, we would defer buffer allocation via the `ReadCallback`). The best thing that I can think of here is to just reserve a small fixed size buffer (~e.g. 512 bytes) and continuously call SSL_read_early_data() to handle it.